bundles/org.eclipse.equinox.simpleconfigurator/src/org/eclipse/equinox/internal/simpleconfigurator/ConfigApplier.java

```diff
@@ -435,49 +416,22 @@ private boolean isFragment(Bundle current) {
 		return (revision != null) && ((revision.getTypes() & BundleRevision.TYPE_FRAGMENT) != 0);
 	}
 
-	private void refreshPackages(Bundle[] bundles, BundleContext context) {
-		if (bundles.length == 0 || packageAdminService == null)
+	private void refreshPackages(Collection<Bundle> bundles) {
+		if (bundles.isEmpty())
 			return;
 
-		// Prior to Luna the Equinox framework would refresh all bundles with the same
-		// BSN automatically.  This is no longer the case for Luna or other framework
-		// implementations.  Here we want to make sure all existing bundles with the
-		// same BSN are refreshed also.
```

Comment:
Is this not relevant anymore?

Reply:
If it would be relevant and not already covered by `getAdditionalRefresh` then it should be handled there ...
But as this codepath is already marked with

`TODO consider removing this case because it can cause inconsistent results.`

I'm not very certain this is worth to worry about.

Reply:
> If it would be relevant and not already covered by `getAdditionalRefresh` then it should be handled there 

Agree.

> ... But as this codepath is already marked with
> 
> `TODO consider removing this case because it can cause inconsistent results.`
> 
> I'm not very certain this is worth to worry about.

If in doubt I would worry about it. But if it is not necessary anymore it should be removed yes, but we should then be certain about that. :)

Reply:
I now added this back even though it seems only there for historic reasons, if one refreshes always all bundles with same BSN then maybe one should also refresh all providers of a package... in the end it then seems better to probably refresh the whole state always as indicated by the comment....

Reply:
> I now added this back even though it seems only there for historic reasons, 

Thanks. Looks better now.

> if one refreshes always all bundles with same BSN then maybe one should also refresh all providers of a package... in the end it then seems better to probably refresh the whole state always as indicated by the comment....

Yes, this could probably be improved or removed. But it should then be done completely and not only partly.